### PR TITLE
Implement login endpoint 

### PR DESCRIPTION
**Pull request by JafarMaash**
## Description
*Please include a summary of the change and which issue number is fixed (e.g. Fixes #1). Please also include relevant motivation and context.*
Implementation for issue #33 
Login endpoint flow:
- Client passes their Google ID token to the login endpoint.
- Token is verified using Google API
- If email is recognised in database, then return custom JWT for later request authentication.
- Otherwise, create new user in database then return the custom JWT.

## How has this been tested?
*Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.*
Due to the time-sensitive nature of Google ID tokens (1 hour before expiry), it is difficult to automatically test this endpoint.
Tested by posting the Team google account's ID token, and decoding the JWT returned. See screenshots for details.

Could add unit tests for the insertion/retrieval of a user in the database in future commits.
## Screenshots
*Provide screenshots of what has been implemented. Leave blank if not applicable*
Returned custom JWT:
image

Postman request to endpoint 
image

## Checklist
*(Leave blank if not applicable)*

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by JafarMaash** [comment]
Note, in order to test POST to an endpoint, CSRF was disabled. This change was not committed as it was done for testing purposes and would be bad practice.

File: src/main/java/com/team701/buddymatcher/controllers/users/UserController.java [hunk 1]
```diff
@@ -56,4 +66,55 @@ public UserController(UserService userService, ModelMapper modelMapper) {
         }
     }
 
+    @Operation(summary = "Post method for a user logging in using Google")
+    @PostMapping(path = "/login", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> loginWithGoogle(HttpServletRequest request) throws GeneralSecurityException, IOException {
+        String CLIENT_ID = "158309441002-q8q49tjicngt1tp6p9t7ecvdrn9ar78j.apps.googleusercontent.com";
+        System.out.println("Made it");
```
Comment (uoa-ypec413) [inline src/main/java/com/team701/buddymatcher/controllers/users/UserController.java]: Could you please remove these System.out.println() statements before merging?

**Review by uoa-ypec413 — APPROVED** [review]
Good job on this! 
Other than the print statements this LGTM :)
Adding unit tests like you stated will be useful, but I agree that it should be pushed back to future issues since this is high priority and needs to be merged ASAP.

File: src/main/java/com/team701/buddymatcher/controllers/users/UserController.java (hunk 1 above)
Comment (JafarMaash) [reply src/main/java/com/team701/buddymatcher/controllers/users/UserController.java]: Went through on a double-check and removed (hopefully) all the print statements :)

Comment (uoa-ypec413) [reply src/main/java/com/team701/buddymatcher/controllers/users/UserController.java]: I found one more, will comment on that one now :)

File: src/main/java/com/team701/buddymatcher/controllers/users/UserController.java [hunk 2]
```diff
@@ -56,4 +66,49 @@ public UserController(UserService userService, ModelMapper modelMapper) {
...
+                user = userService.retrieveByEmail(email);
+            }
+                // return custom JWT
+            JwtTokenUtil util = new JwtTokenUtil();
+            String token = util.generateToken(user);
+            return new ResponseEntity<>(token, HttpStatus.OK);
+
+
+        } else {
+            System.out.println("Invalid ID token.");
```
Comment (uoa-ypec413) [inline src/main/java/com/team701/buddymatcher/controllers/users/UserController.java:107]: @JafarMaash there's still a system out here :)

**Pull request merged**